File: packages/ts/src/application/ports/id-generator.ts
import type { RequestId, RunId, ToolCallId } from "../../domain/identifiers";

/** Produces the branded identifiers used across a run. Injected for determinism. */
export interface IdGenerator {
  runId(): RunId;
  requestId(): RequestId;
  toolCallId(): ToolCallId;
}

File: packages/py/src/airlock/application/ports/id_generator.py
"""IdGenerator port: the identifiers the agent mints (run and request ids).

Tool-call ids come from the model, not from here."""

from typing import Protocol

from airlock.domain.identifiers import RequestId, RunId


class IdGenerator(Protocol):
    def run_id(self) -> RunId: ...

    def request_id(self) -> RequestId: ...
